cmake/API.cmake

```diff
@@ -301,7 +301,15 @@ function(register_fprime_executable)
     endif()
     get_nearest_build_root(${CMAKE_CURRENT_LIST_DIR})
     # Explicit call to module register
-    generate_executable("${EX_NAME}" "${SC_IFS}" "${MD_IFS}")
+    generate_executable("${EX_NAME}_exe" "${SC_IFS}" "${MD_IFS}")
+    set_target_properties("${EX_NAME}_exe" PROPERTIES OUTPUT_NAME "${EX_NAME}")
+    add_custom_target(${EX_NAME} ALL)
+    add_dependencies("${EX_NAME}" "${EX_NAME}_exe")
+
+    # Only install into artifacts directory in release builds when SKIP_INSTALL is not set.
+    if (NOT DEFINED SKIP_INSTALL AND CMAKE_BUILD_TYPE STREQUAL "RELEASE")
```

Comment:
This implies that for the "Testing" build, you will not make the installed directory?  Thus how do I run the gds against a testing build?

Reply:
Oh, interesting. I figured the only use of testing builds was unit tests, but I guess there's also the use case of debug binaries. I want to be certain that people don't mix release and debug builds. We could do somethign like `build-artifacts/<platform>/<RELEASE|TESTING>/...` for organization. It adds a new directory level to the build artifacts but does separate debug from release builds.

Is adding that additional complexity better than making users dig into the cmake build directory to find the debug binary? I'd say that the usage debug builds are rare enough that it's probably not worth adding another directory level. It could also cause user confusion because the RELEASE and TESTING directories are built separately, so they can get out of sync.